### PR TITLE
Auth functionality

### PR DESCRIPTION
**Pull request by sameep-s**
What's added:

- Auth(login, signup , logout)
- FIlters

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
       <Routes>
         <Route path="/" element={<LandingPage />} />
-        <Route path="/profile" element={<Profile />} />
-        <Route path="/notes" element={<Notes />} />
-        <Route path="/archives" element={<Archives />} />
-        <Route path="/trash" element={<Trash />} />
+        <Route path="/signup" element={<Signup />} />
+        <Route path="/login" element={<Login />} />
+
+
```
Comment (arjunbharti774) [inline src/App.js:12]: avoid extra spacing if not necessary

File: src/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const signupUser = async (email, password, confirmPassword, firstName, lastName, age) => {
+        try {
+            const { status, data: { encodedToken, createdUser } } = await signupService(email, password, confirmPassword, firstName, lastName, age);
+
+            if (status === 201) {
+                localStorage.setItem('tokenNotes', encodedToken);
+                localStorage.setItem('userNotes', JSON.stringify(createdUser));
+                setEncodedToken(encodedToken);
+                setUser(createdUser);
+                console.log(`created user`, createdUser);
```
Comment (arjunbharti774) [inline src/context/auth-context.js:50]: remove logs

**Review by arjunbharti774 — APPROVED** [review]
improvements:

1. avoid extra spacing the code
2. remove console logs

Really liked file structure and clean coding practice in the context files, well substituted code and ES6 syntax

File: src/App.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+        <Route path="/profile" element={<RequireAuth> <Profile /></RequireAuth>} />
+        <Route path="/notes" element={<RequireAuth> <Notes /></RequireAuth>} />
+        <Route path="/archives" element={<RequireAuth> <Archives /></RequireAuth>} />
+        <Route path="/trash" element={<RequireAuth> <Trash /></RequireAuth>} />
       </Routes>
```
Comment (rahulyadav139) [inline src/App.js:17]: suggestion: you can use 'outlet' feature provided by react-router for protecting private routes.

File: src/pages/Login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        loginUser(loginFormData.email, loginFormData.password);
+        if (user) {
+            navigate("/", { replace: true });
+        }
+
+        setLoginFormData(() => ({ email: "", password: "" }));
+    }
+
+    const testLoginHandler = () => {
+        loginUser("sam@gmail.com", "s123");
```
Comment (rahulyadav139) [inline src/pages/Login/Login.jsx:38]: suggestion: add test login credential into .env file.

**Review by rahulyadav139 — APPROVED** [review]
Appreciation: everything is looking great. Codes are readable. Great work.

File: src/App.js (hunk 3 above)
Comment (sameep-s) [reply src/App.js:17]: Noted, will look into that.

File: src/App.js (hunk 1 above)
Comment (sameep-s) [reply src/App.js:12]: will remove it.

File: src/context/auth-context.js (hunk 2 above)
Comment (sameep-s) [reply src/context/auth-context.js:50]: will do.

**Pull request merged**